### DISCUSSION TITLE
Get magic Link to work on safari on the iphone + gmail app?

### DISCUSSION THREAD

**Discussion by Justinfan827** [supabase/supabase#15708] [Questions]
Hey y'all! I realize that links can only be used from the same browser that they are sent on. With this in mind, how can I get magic links to work when the user is opening the link from the gmail app? It seems to always open the link in the in app browser.

From the docs: https://supabase.com/docs/guides/auth/auth-magic-link

> When making use of Magic Links with [the PKCE Flow](https://supabase.com/blog/supabase-auth-sso-pkce#introducing-pkce) do bear in mind that links can only be used in the same browser that they are sent from. Consequently, a magic link sent from Chrome on Desktop will be invalid if used on a Mobile Device.

![IMG_1337](https://github.com/supabase/supabase/assets/22778148/068db311-3ea2-437d-9302-3b9991e12079)

![IMG_1338](https://github.com/supabase/supabase/assets/22778148/c5083413-e708-4dcc-8f0e-e2bf3c3e55f5)

**Comment by Ryan-Ouyang** [comment]
Running into this issue as well - it also doesn't seem like there's a way to force the link to open in the native default browser, which makes logging users in a huge problem: we are basically forced into having users copy and paste the link from their email into their default browser.

**Comment by tim-vyg** [comment]
Experiencing the same problem, any solutions / hack-arounds?

**Comment by virgildotcodes** [comment]
Encountering the same issue here!

**Reply by tim-vyg** [reply]
Think we're gonna have to send a token as well which they can input if the link doesn't work -> https://supabase.com/docs/reference/javascript/auth-verifyotp

**Comment by wilg** [comment]
Unrelated to Supabase but I wrote up some ideas for how to deal with this here, as well as some information on the user flows. https://github.com/abevoelker/devise-passwordless/issues/61

**Comment by AndrewHenderson** [comment]
Any movement on this or new ideas for workarounds? I'm still experiencing this issue.

**Comment by mike-altonji** [comment]
Does this limitation also exist in Firsebase? Considering migrating my auth there to avoid the issue. I like the idea of magic links, but this isn't functional
